packages/babel-plugin-transform-runtime/scripts/build-dist.js

```diff
@@ -92,51 +92,58 @@ function writeCorejsExports(pkgDirname, runtimeRoot, paths) {
   outputFile(pkgJsonPath, JSON.stringify(pkgJson, undefined, 2) + "\n");
 }
 
-function writeHelpers(runtimeName, { corejs } = {}) {
-  const helperPaths = writeHelperFiles(runtimeName, { corejs, esm: false });
-  const helperESMPaths = writeHelperFiles(runtimeName, { corejs, esm: true });
-  writeHelperExports(runtimeName, helperPaths.concat(helperESMPaths));
+function writeHelperFile(
+  runtimeName,
+  pkgDirname,
+  helperPath,
+  helperName,
+  { esm, corejs }
+) {
+  const filePath = path.join(helperPath, esm ? "index.mjs" : "index.js");
+  const fullPath = path.join(pkgDirname, filePath);
+
+  outputFile(
+    fullPath,
+    buildHelper(runtimeName, pkgDirname, fullPath, helperName, { esm, corejs })
+  );
+
+  return `./${filePath}`;
 }
 
-function writeHelperExports(runtimeName, helperPaths) {
+function writeHelpers(runtimeName, { corejs } = {}) {
+  const pkgDirname = getRuntimeRoot(runtimeName);
   const helperSubExports = {};
-  for (const helperPath of helperPaths) {
-    helperSubExports[helperPath.replace(".js", "")] = helperPath;
+  for (const helperName of helpers.list) {
+    const helperPath = path.join("helpers", helperName);
+    helperSubExports[`./${helperPath}`] = {
+      import: writeHelperFile(runtimeName, pkgDirname, helperPath, helperName, {
```

Comment:
for the ideal experience, the very same value should be repeated for a priority `"module"` condition that is supported by webpack (and will most likely be supported by Rollup as well), in node `require` cant load ESM so it can lead to the same module being loaded twice (once for ESM loader and once for the CJS loader) which ain't ideal for the web consumers so this condition allows the same file (authored in ESM) to be loaded for both loaders, so this behaves pretty much as the `package.json#module`